hal/src/nRF52840/ble_hal.cpp

```diff
@@ -2206,9 +2197,9 @@ BleObject::ConnectionsManager::BleConnection* BleObject::ConnectionsManager::fet
     return nullptr;
 }
 
-int BleObject::ConnectionsManager::addConnection(const BleConnection& connection) {
+int BleObject::ConnectionsManager::addConnection(BleConnection& connection) {
```

Comment:
This is confusing semantically. This method intends to take ownership of `connection`, so it should take it by rvalue reference: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f18-for-will-move-from-parameters-pass-by-x-and-stdmove-the-parameter